.yamato/scripts/update_from_upstream.sh

```diff
@@ -3,11 +3,13 @@
 # Prequisites:
 # This script assumes git and github CLR (gh) are installed.
 
-# Fetch from upstream and rebase our local main on it
+# Cause the script to fail if any individual command fails
+set -e
+
+# Fetch from upstream and to a fast forward merge into our local main
 git remote add upstream https://github.com/dotnet/runtime
-git fetch upstream
-git checkout --track origin/main
-git rebase upstream/main
+git fetch upstream main
```

Comment:
@joncham It looks like we need the previous set of Git commands here, otherwise we will end up in a detached head state that won't work later.